openlibrary/plugins/upstream/code.py

```diff
@@ -94,12 +94,15 @@ def GET(self):
 
 
 class merge_work(delegate.page):
-    path = r"(/works/OL\d+W)/merge"
+    path = "/works/merge"
 
-    def GET(self, key):
-        return "This looks like a good place for a merge UI!"
+    def GET(self):
+        user = web.ctx.site.get_user()
+        if not user or (not user.is_admin() and not user.is_librarian()):
+            raise web.HTTPError('403 Forbidden')
+        return render_template('merge/works')
 
-    def POST(self, key):
+    def POST(self):
         pass
```

Comment:
why have a POST for this if the action is to pass?

Reply:
RM"d